--- binutils.orig/bfd/configure	2019-06-24 14:37:35.984087086 +0100
+++ binutils-2.32/bfd/configure	2019-06-24 17:32:52.515541752 +0100
@@ -18865,11 +18865,11 @@ _LT_EOF
       esac
     done ;;
     "bfd_stdint.h":C)
-if test "$GCC" = yes; then
-  echo "/* generated for " `$CC --version | sed 1q` "*/" > tmp-stdint.h
-else
-  echo "/* generated for $CC */" > tmp-stdint.h
-fi
+
+
+
+
+
 
 sed 's/^ *//' >> tmp-stdint.h <<EOF
 
